api/controllers/v1/ncn_test.go

```diff
@@ -90,3 +90,75 @@ func TestNcnCreateRebuildWorkflow(t *testing.T) {
 		assert.Equal(t, http.StatusBadRequest, res.Code)
 	})
 }
+
+func TestNcnsCreateRebuildWorkflow(t *testing.T) {
+
+	gin.SetMode(gin.TestMode)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	executeWithContext := func(
+		workflowService *mocks.MockWorkflowService,
+		requestBody string,
+	) *httptest.ResponseRecorder {
+		response := httptest.NewRecorder()
+		context, ginEngine := gin.CreateTestContext(response)
+
+		requestUrl := "/v1/ncns/rebuild"
+
+		context.Request, _ = http.NewRequest("POST", requestUrl, strings.NewReader(requestBody))
+
+		ginEngine.POST("/v1/ncns/rebuild", NewNcnController(workflowService, *utils.GetLogger().GetGinLogger().Logger).NcnsCreateRebuildWorkflow)
+		ginEngine.ServeHTTP(response, context.Request)
+		return response
+	}
+
+	t.Run("Happy", func(t *testing.T) {
+
+		workflowServiceMock := mocks.NewMockWorkflowService(ctrl)
+		workflowServiceMock.EXPECT().CreateRebuildWorkflow(gomock.Any()).Return(
+			&v1alpha1.Workflow{
+				ObjectMeta: v1.ObjectMeta{Name: "mocked", Labels: map[string]string{"targetNcn": "mocked-target-ncn"}},
+			}, nil)
+		res := executeWithContext(workflowServiceMock, `{
+			"hosts": [
+			  "ncn-w003",    "ncn-w003",    "ncn-w003",    "ncn-w003"
+			]
+		  }`)
+		assert.Equal(t, http.StatusOK, res.Code)
+	})
+
+	t.Run("Error", func(t *testing.T) {
+
+		workflowServiceMock := mocks.NewMockWorkflowService(ctrl)
+		workflowServiceMock.EXPECT().CreateRebuildWorkflow(gomock.Any()).Return(nil, fmt.Errorf("mocked error"))
+		res := executeWithContext(workflowServiceMock, `{
+			"hosts": [
+			  "ncn-w003",    "ncn-w003",    "ncn-w003",    "ncn-w003"
+			]
+		  }`)
+		assert.Equal(t, http.StatusInternalServerError, res.Code)
+	})
+
+	t.Run("wrong hostname - invalid", func(t *testing.T) {
+
+		workflowServiceMock := mocks.NewMockWorkflowService(ctrl)
+		res := executeWithContext(workflowServiceMock, `{
+			"hosts": [
+			  "ncn-s003",    "ncn-m003",    "ncn-w003",    "ncn-w003"
+			]
+		  }`)
+		assert.Equal(t, http.StatusBadRequest, res.Code)
+	})
+
+	t.Run("invalid request", func(t *testing.T) {
+
+		workflowServiceMock := mocks.NewMockWorkflowService(ctrl)
+		res := executeWithContext(workflowServiceMock, `{
+			"hosts": [
```

Comment:
Since this doesn't span multiple lines it should be condensed onto one line instead of 3.